d2layouts/d2grid/layout.go

```diff
@@ -126,6 +122,34 @@ func withoutGridDiagrams(ctx context.Context, g *d2graph.Graph, layout d2graph.L
 			}
 			if obj.LabelDimensions.Height != 0 {
 				labelHeight := float64(obj.LabelDimensions.Height) + 2*label.PADDING
+
+				{
+					// also check for grid cells with outside top labels
+					topY := gd.objects[0].TopLeft.Y
```

Comment:
objects[0] guaranteed to exist? even so, rather u make negative infinite

Reply:
added a comment that it always exists and that we specifically want the first object's y here